homework 2 is done

pachinaanna opened the pull request.

pachinaanna committed:
homework 2 is done

shoggy commented on pom.xml:
```diff
@@ -0,0 +1,76 @@
+                <configuration>
+                    <source>1.8</source>
+                    <target>1.8</target>
+                    <encoding>UTF-8</encoding>
```
Discussed in PR for HW1
```
<properties>
    <project.build.sourceEncoding>UTF-8
    <maven.compiler.source>1.8
    <maven.compiler.target>1.8
```

shoggy commented on pom.xml:
```diff
@@ -0,0 +1,76 @@
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-java</artifactId>
+            <version>3.141.59</version>
```
Please move version to `properties` and add `<scope>test`

shoggy commented on pom.xml:
```diff
@@ -0,0 +1,76 @@
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-simple</artifactId>
+            <version>1.6.2</version>
```
This version too

shoggy commented on src/test/java/com/epam/trn/hw2/AbstractTest.java:
```diff
@@ -0,0 +1,48 @@
+    @BeforeMethod
+    public void setUp() {
+
+        System.setProperty("webdriver.chrome.driver", "c://Drivers//chromedriver.exe");
```
In a real project, you may have an agreement about driver placement.
But in our case would be better to use bonigarcia:webdrivermanager, or at least place driver somewhere in the project

shoggy commented on src/test/java/com/epam/trn/hw2/AbstractTest.java:
```diff
@@ -0,0 +1,48 @@
+        webDriver = new ChromeDriver();
+        webDriver.manage().timeouts().implicitlyWait(10000, TimeUnit.MILLISECONDS);
+        webDriver.manage().timeouts().pageLoadTimeout(20000, TimeUnit.MILLISECONDS);
+        webDriver.manage().timeouts().setScriptTimeout(30000, TimeUnit.MILLISECONDS);
```
What is the reason for overriding default values?

shoggy commented on src/test/java/com/epam/trn/hw2/AbstractTest.java:
```diff
@@ -0,0 +1,48 @@
+        webDriver.manage().timeouts().implicitlyWait(10000, TimeUnit.MILLISECONDS);
+        webDriver.manage().timeouts().pageLoadTimeout(20000, TimeUnit.MILLISECONDS);
+        webDriver.manage().timeouts().setScriptTimeout(30000, TimeUnit.MILLISECONDS);
+        webDriver.manage().window().maximize();
```
Let's try to start the browser already maximized. `ChromeOptions` is the key

shoggy commented on src/test/java/com/epam/trn/hw2/ex1/Exercise1.java:
```diff
@@ -0,0 +1,147 @@
+        // Data: "HOME", "CONTACT FORM", "SERVICE", "METALS & COLORS"
+        // Exp. results: Menu buttons are displayed and have proper texts
+
+        List<WebElement> buttons = webDriver.findElements(By.xpath("//ul[contains(@class, 'm-l8')]/li"));
```
```suggestion
        List<WebElement> buttons = webDriver.findElements(By.css("ul.m-l8 > li"));
```
Same for other locators

pachinaanna committed:
pom.xml - new dependencies added, versions moved
AbstractTest - webdrivermanager, ChromeOptions
Exercise1, Exercise2 - some locators changed

pachinaanna committed:
Merge branch 'master' into homework_2
